### ISSUE TITLE
DeferredList Documentation Lies!

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3323]
|<img alt="nathan.stocks's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| nathan.stocks reported|
|-|-|
|Trac ID|trac#3323|
|Type|defect|
|Created|2008-06-30 21:17:22Z|
|Branch|https://github.com/twisted/twisted/tree/deferredlist-docs-3323|

This section on DeferredList's is incorrect:

http://twistedmatrix.com/projects/core/documentation/howto/defer.html#auto7

From the section of the above page:

```
> # At this point, dl will fire its callback, printing:
> #     [(1, 'one'), (0, 'bang!'), (1, 'three')]
> # (note that defer.SUCCESS == 1, and defer.FAILURE == 0)
> 
> A standard DeferredList will never call errback.
```

Here's the actual output when run in Twisted 8.0.1:

```
$ python test_DeferredList.py 
/Library/Python/2.5/site-packages/twisted/internet/defer.py:262: DeprecationWarning: Don't pass strings (like 'bang!') to failure.Failure (replacing with a DefaultException).
  fail = failure.Failure(fail)
[(True, 'one'), (False, <twisted.python.failure.Failure <class 'twisted.python.failure.DefaultException'>>), (True, 'three')]
Unhandled error in Deferred:
Traceback (most recent call last):
Failure: twisted.python.failure.DefaultException: bang!
```

I see three problems:

  1. The deprecated behaviour.  (Passing a string to an errback?)
  2. Output doesn't match.  (defer.SUCCESS and defer.FAILURE are True and False, respectively, not 1 and 0)
  3. The docs say "A standard DeferredList will never call errback.", but it appears that an errback was actually called.

Attachments:

* [test_DeferredList.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/502/5023a214790a3c658c36e93372cd7feb7546d279/6dc95c62e488d54de53131e2d3187ac4aefdfd75.py) (486 bytes) - added by nathan.stocks on 2008-06-30 21:23:53Z - A file that has the sample code in question in a runnable form.

<details><summary>Searchable metadata</summary>

```
trac-id__3323 3323
type__defect defect
reporter__nathan_stocks nathan.stocks
priority__normal normal
milestone__ 
branch__branches_deferredlist_docs_3323 branches/deferredlist-docs-3323
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1214860642000000 1214860642000000
changetime__1215015281000000 1215015281000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

fixing description markup

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24188]) Branching to 'deferredlist-docs-3323'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [24189]) Correct the example (with output) for DeferredList

refs [#3323](https://github.com/twisted/twisted/issues/3323)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="nathan.stocks's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|nathan.stocks commented|
|-|-|

It appears that the third concern I listed is a misunderstanding on my part.  The statement _"A standard DeferredList will never call errback."_ from the documentation does not mean that errors from Deferreds in the DeferredList will be consumed, it only literally means that errback is never called on the DeferredList itself.

It may be helpful to simply state earlier in the documentation something to the effect of "you still need to handle errors in your Deferreds by adding errbacks to them, unless you are going to use the consumeErrors option talked about later."

The other two concerns still stand.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_6"></a>|@jerub set owner to @exarkun|
|-|-|

I've tested this code manually in a python interpeter, and am happy with the code listing and the updated paragraph. I think this can be merged.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [24216]) Merge deferredlist-docs-3323

Author: exarkun
Reviewer: jerub
Fixes: [#3323](https://github.com/twisted/twisted/issues/3323)

Change some of the `DeferredList` documentation in the deferred
howto so that it doesn't use deprecated features (string
exceptions), notes that `SUCCESS` and `FAILURE` are booleans now,
not boolean-style integers, and uses and mentions the
`consumeErrors` feature of `DeferredList` to avoid putting a
confusing failure report onto stdout.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
